metagraph/src/cli/load/load_annotated_graph.cpp

```diff
@@ -22,7 +22,7 @@ using mtg::common::logger;
 std::unique_ptr<AnnotatedDBG> initialize_annotated_dbg(std::shared_ptr<DeBruijnGraph> graph,
                                                        const Config &config) {
     // TODO: check and wrap into canonical only if the graph is primary
-    if (config.canonical && !graph->is_canonical_mode())
+    if (config.canonical && !graph->is_canonical_mode() && config.identity != Config::ASSEMBLE)
```

Comment:
why not for assembly?

Normally you just don't pass `--canonical` to assembly, and this condition won't be satisfied anyway, right?

Reply:
True, but metagraph will still run if there's a stray `--canonical` flag in the `ASSEMBLE` regime.

Maybe instead we can have metagraph print a warning in `config.cpp` if this case happens to warn that it'll be slower and add extra k-mers to the output.

Reply:
or we could print the warning here

Reply:
better unset it in config.cpp, so it's always false for ASSEMBLY.
Or just leave as it was. `--canonical` is not listed in usage for ASSEMBLY, so if pass it, the behavior is undefined.